src/main/java/org/opensearch/timeseries/breaker/MemoryCircuitBreakerNumericSetting.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.timeseries.breaker;
+
+import static java.util.Collections.unmodifiableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.timeseries.settings.DynamicNumericSetting;
+
+public class MemoryCircuitBreakerNumericSetting extends DynamicNumericSetting {
```

Comment:
Can you move your code to ADNumericSetting ? We don't need to create a new class for each setting. Also, we covered setting init in TimeSeriesAnalyticsPlugin. Once you moved your code to ADNumericSetting, we don't need another call to initialize memory circuit breaker setting.

Reply:
moved the logic to `ADNumericSetting` file in the latest revision